packages/protocol/contracts/layer2/DelegateOwner.sol

```diff
@@ -45,54 +38,29 @@ contract DelegateOwner is EssentialContract, IMessageInvocable {
         uint64 indexed txId, address indexed target, bool isDelegateCall, bytes txdata
     );
 
-    /// @notice Emitted when the admin has been changed.
-    /// @param oldAdmin The old admin address.
-    /// @param newAdmin The new admin address.
-    event AdminUpdated(address indexed oldAdmin, address indexed newAdmin);
-
     error DO_DRYRUN_SUCCEEDED();
-    error DO_INVALID_PARAM();
     error DO_INVALID_SENDER();
     error DO_INVALID_TARGET();
     error DO_INVALID_TX_ID();
     error DO_PERMISSION_DENIED();
 
-    modifier onlyAdminOrRemoteOwner() {
-        if (!_isAdminOrRemoteOwner(msg.sender)) revert DO_PERMISSION_DENIED();
-        _;
+    constructor(uint64 _l1ChainId, address _l1Bridge, address _daoController) EssentialContract() {
+        l1ChainId = _l1ChainId;
+        l1Bridge = _l1Bridge;
+        daoController = _daoController;
     }
 
-    constructor(address _bridge) EssentialContract() {
-        bridge = _bridge;
-    }
-
-    /// @notice Initializes the contract.
-    /// @param _remoteOwner The real owner on L1 that can send a cross-chain message to invoke
-    /// `onMessageInvocation`.
-    /// @param _remoteChainId The L1 chain's ID.
-    /// @param _admin The admin address.
-    function init(
-        address _remoteOwner,
-        uint64 _remoteChainId,
-        address _admin
-    )
-        external
-        initializer
-    {
-        // This contract's owner will be itself.
+    function init() external initializer {
         __Essential_init(address(this));
-
-        require(_remoteOwner != address(0), DO_INVALID_PARAM());
-        require(_remoteChainId != 0, DO_INVALID_PARAM());
-        require(_remoteChainId != block.chainid, DO_INVALID_PARAM());
-
-        remoteChainId = _remoteChainId;
-        remoteOwner = _remoteOwner;
-        admin = _admin;
     }
 
     /// @inheritdoc IMessageInvocable
-    function onMessageInvocation(bytes calldata _data) external payable onlyAdminOrRemoteOwner {
+    function onMessageInvocation(bytes calldata _data) external payable {
+        require(msg.sender == l1Bridge, DO_INVALID_SENDER());
```

Comment:
imo here should be `msg.sender == l2Bridge`?

Reply:
This is a bug!
We should also have l2Bridge immutable @adaki2004